eva-pipeline/src/main/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotGenerateInputBatch.java

```diff
@@ -0,0 +1,190 @@
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Dump a list of variants without annotations from mongo
+ *
+ * Step class that:
+ * - READ: read the variants without annotations from mongo
+ * - PROCESS convert, filter, validate... the {@link VariantWrapper}
+ * - LOAD: write the {@link VariantWrapper} into a flatfile
+ *
+ * TODO:
+ * - Handle the overwrite
+ * - The variant list should be compressed. It is not possible to write into a zipped file with FlatFile item writer
+ *  see jmmut comment at https://github.com/EBIvariation/eva-v2/pull/22
+ *  We can create an extra step to convert the file and remove the nonp-zipped one
+ *  https://www.mkyong.com/java/how-to-compress-a-file-in-gzip-format/
+ *  https://examples.javacodegeeks.com/core-java/io/fileinputstream/compress-a-file-in-gzip-format-in-java/
+ *  http://www.journaldev.com/966/java-gzip-example-compress-and-decompress-file-in-gzip-format-in-java
+ */
+
+@Configuration
+@EnableBatchProcessing
+@Import(VariantJobArgsConfig.class)
+public class VariantsAnnotGenerateInputBatch {
+
+    private static final Logger logger = LoggerFactory.getLogger(VariantsAnnotGenerateInputBatch.class);
+
+    @Autowired
+    private StepBuilderFactory steps;
+
+    @Autowired
+    private ObjectMap pipelineOptions;
+
+/*
+    public static final String jobName = "variantsAnnotGenerateInputJob";
+
+    @Bean
+    public Step variantsAnnotGenerateInputBatchStep(ItemReader<DBObject> reader,
+                      ItemProcessor<DBObject, VepInputLine> processor,
+                      ItemWriter<VepInputLine> writer) {
+        return steps.get("step1").<DBObject, VepInputLine> chunk(10)
+                .reader(reader)
+                .processor(processor)
+                .writer(writer).allowStartIfComplete(false)
+                .build();
+    }*/
+
+    @Bean
+    @Qualifier("variantsAnnotGenerateInputBatchStep")
+    public Step variantsAnnotGenerateInputBatchStep() throws Exception {
+        return steps.get("variantsAnnotGenerateInputBatchStep").<DBObject, VariantWrapper> chunk(10)
+                .reader(variantReader())
+                .processor(vepInputLineProcessor())
+                .writer(vepInputWriter()).allowStartIfComplete(false)
+                .build();
+    }
+
+    @Bean
+    public ItemReader<DBObject> variantReader() throws Exception {
+        return initReader(pipelineOptions.getString("dbCollectionVariantsName"), mongoOperations());
```

Comment:
I was using them in test.. that's why but yes I agree not very nice to have. I removed the public method and re-worked a bit the test as well